trace-mini-agent/src/trace_flusher.rs

```diff
@@ -25,29 +24,45 @@ pub struct ServerlessTraceFlusher {}
 #[async_trait]
 impl TraceFlusher for ServerlessTraceFlusher {
     async fn start_trace_flusher(&self, mut rx: Receiver<pb::TracerPayload>) {
-        let mut buffer: Vec<pb::TracerPayload> = Vec::with_capacity(BUFFER_FLUSH_SIZE);
-
-        // receive trace payloads from http endpoint handlers and add them to the buffer. flush if
-        // the buffer gets to BUFFER_FLUSH_SIZE size.
-        while let Some(tracer_payload) = rx.recv().await {
-            buffer.push(tracer_payload);
-            if buffer.len() >= BUFFER_FLUSH_SIZE {
-                self.flush_traces(buffer.to_vec()).await;
-                buffer.clear();
+        let buffer: Arc<Mutex<Vec<pb::TracerPayload>>> = Arc::new(Mutex::new(Vec::new()));
+
+        let buffer_producer = buffer.clone();
+        let buffer_consumer = buffer.clone();
+
+        tokio::spawn(async move {
+            while let Some(tracer_payload) = rx.recv().await {
+                let mut buffer = buffer_producer.lock().await;
+                buffer.push(tracer_payload);
             }
+        });
+
+        loop {
+            tokio::time::sleep(time::Duration::from_millis(3000)).await;
+
+            let mut buffer = buffer_consumer.lock().await;
+            if buffer.is_empty() {
+                continue;
```

Comment:
you could remove the continue by reversing the condition and move flush + clear in the block?

Reply:
done